lib/thesis/api_controller.ex

```diff
@@ -18,6 +18,30 @@ defmodule Thesis.ApiController do
     json conn, %{}
   end
 
+  def upload_file(conn, %{"file" => ""}), do: json conn, %{path: ""}
+  def upload_file(conn, %{"file" => file}) do
+    path = uploader.upload(file)
+    json conn, %{path: path}
+  end
+  def upload_file(conn, _), do: json conn, %{path: ""}
```

Comment:
Controller actions handle the file upload using the specified uploader. The default uploader is RepoUploader but you can create your own and specify in the config. The requirement is 1 function - `upload/1` - the ability to handle a `%Plug.Upload{}` struct and return a string.